### PR TITLE
`bevy_color`: Add `Laba` to support Lab Color Model

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Improve compatibility with CSS Module 4
- Simplify `Lcha` conversion functions

## Solution

- Added `Laba` which implements the Lab color model.
- Updated `Color` and `LegacyColor` accordingly.

## Migration Guide

- Convert `Laba` to either `Xyza` or `Lcha` using the provided `From` implementations and then handle accordingly.

## Notes

The Lab color space is a required stepping stone when converting between XYZ and Lch, therefore we already use the Lab color model, just in an nameless fashion prone to errors.

This PR also includes a slightly broader refactor of the `From` implementations between the various colour spaces to better reflect the graph of definitions. My goal was to keep domain specific knowledge of each colour space contained to their respective files (e.g., the `From<Oklaba> for LinearRgba` definition was in `linear_rgba.rs` when it probably belongs in `oklaba.rs`, since Linear sRGB is a fundamental space and Oklab is defined in its relation to it)

### CODE REVIEW

File: crates/bevy_color/src/color.rs [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
-    Hsla(Hsla),
-    /// A color in the LCH color space with alpha.
-    Lcha(Lcha),
-    /// A color in the Oklaba color space with alpha.
-    Oklaba(Oklaba),
-    /// A color in the XYZ color space with alpha.
-    Xyza(Xyza),
+use macros::define_color;
+
+define_color! {
```
Comment (alice-i-cecile) [inline crates/bevy_color/src/color.rs:7]: This macro usage is going to be controversial. I personally think we should revert it, both in the interests of not getting bogged down and in terms of maintainability.

Comment (bushrat011899) [reply crates/bevy_color/src/color.rs:7]: That's part of the reason why I put this up as a draft now, just to get feedback on this shortcut. I'm happy to remove both macros

**Comment by alice-i-cecile** [comment]
The prerequisite PR has been merged :) Please rebase when you get a chance and we can add this in.

**Review by alice-i-cecile — APPROVED** [review]

**Review by viridia — APPROVED** [review]
Do we want to explain about derived conversions in the crate-level docs?

**Comment by viridia** [comment]
Also, I know we've been trying to explain to new users what all these various color spaces are *for* but at some point we will have to give up, because the distinctions are (a) hair-splitting and (b) historical, which makes it hard to write a clear and distinct explanation.

**Comment by bushrat011899** [comment]
I was thinking it might be nice to actually include a diagram showing how the various colour spaces relate to each other, but I'm not sure what the best way to include that in Rust Docs would be.

Example Diagram

**Comment by bushrat011899** [comment]
That's definitely the route I'd want to take. I'll make a little PR for this now

**Pull request merged**